components/camel-test/camel-test-junit5/src/main/java/org/apache/camel/test/junit5/properties/TestPropertiesSource.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.test.junit5.properties;
+
+import java.lang.reflect.Field;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.RuntimeCamelException;
+
+import static org.junit.platform.commons.util.ReflectionUtils.makeAccessible;
+
+public class TestPropertiesSource implements org.apache.camel.spi.PropertiesSource {
```

Comment:
This would break CKC test framework, but I think I can reimplement the logic on `getProperty` with the same pattern there. So, IMHO, it would not be a problem.